### ISSUE TITLE
Support size and axis argument for `jnp.unique`

### ISSUE THREAD

**Issue by n-gao** [google/jax#7090]
Please:

- [x] Check for duplicate requests.
- [x] Describe your goal, and if possible provide a code snippet with a motivating example.

Hi,
at the moment it is not possible to have `jnp.unique` functioning in jit with `axis` being something else than `None`. While the `size` argument is supposed to make `jnp.unique` jit compatible, it doesn't work with also the `axis` argument being set.

A simple use case for this would be to find the unique vectors in a set.

**Comment by jakevdp** [comment]
Thanks - this is being tracked in #3614
